run-build-functions.sh

```diff
@@ -452,16 +452,24 @@ install_dependencies() {
     local goVersion=$(cat .go-version)
     if [ "$installGoVersion" != "$goVersion" ]
     then
-	installGoVersion="$goVersion"
+      installGoVersion="$goVersion"
     fi
   fi
 
   if [ "$GIMME_GO_VERSION" != "$installGoVersion" ]
   then
-    echo "Installing Go version $goVersion"
-    eval "$(GIMME_ENV_PREFIX=$HOME/.gimme_cache/env GIMME_VERSION_PREFIX=$HOME/.gimme_cache/versions gimme $goVersion)"
+    echo "Installing Go version $installGoVersion"
+    GIMME_ENV_PREFIX=$HOME/.gimme_cache/env GIMME_VERSION_PREFIX=$HOME/.gimme_cache/versions gimme $installGoVersion
+    if [ $? -eq 0 ]
+    then
+      source $HOME/.gimme_cache/env/go$installGoVersion.linux.amd64.env
+    else
+      echo "Failed to install Go version '$isntallGoVersion'"
+      exit 1
+    fi
   else
-    eval "$(gimme)"
+    gimme
```

Comment:
Can this fail? If so, we should add a `if ! gimme, then, fi` block.

Reply:
it should be a noop in network, but I'm sure it could fail in some other way. I added a check.